ambry-api/src/main/java/com/github/ambry/store/Store.java

```diff
@@ -171,6 +171,13 @@
    */
   boolean recoverFromDecommission();
 
+  /**
+   * @return {@code true} if the store is disabled on error (due to disk I/O issue).
+   */
+  default boolean disabledOnError() {
```

Comment:
No specific reason. I was trying to avoid adding this new method to other implementations. Maybe I should add it to other impl and throw exception there to make it consistent with other methods in this interface.